packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -0,0 +1,294 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+import Signin from '.';
+import {
+  isOAuthIntegration,
+  isSyncDesktopV3Integration,
+  useAuthClient,
+} from '../../models';
+import { MozServices } from '../../lib/types';
+import { useValidatedQueryParams } from '../../lib/hooks/useValidate';
+import { SigninQueryParams } from '../../models/pages/signin';
+import { useCallback, useEffect, useState } from 'react';
+import firefox from '../../lib/channels/firefox';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { currentAccount } from '../../lib/cache';
+import { useMutation, useQuery } from '@apollo/client';
+import { AVATAR_QUERY, BEGIN_SIGNIN_MUTATION } from './gql';
+import { hardNavigateToContentServer } from 'fxa-react/lib/utils';
+import {
+  RecoveryEmailStatusResponse,
+  AvatarResponse,
+  BeginSigninHandler,
+  BeginSigninResponse,
+  BeginSigninResultError,
+  CachedSigninHandler,
+  LocationState,
+  SigninContainerIntegration,
+} from './interfaces';
+import { getCredentials } from 'fxa-auth-client/browser';
+import { GraphQLError } from 'graphql';
+import {
+  AuthUiErrorNos,
+  AuthUiErrors,
+  composeAuthUiErrorTranslationId,
+} from '../../lib/auth-errors/auth-errors';
+import VerificationMethods from '../../constants/verification-methods';
+import VerificationReasons from '../../constants/verification-reasons';
+import AuthenticationMethods from '../../constants/authentication-methods';
+
+/*
+ * In content-server, the `email` param is optional. If it's provided, we
+ * check against it to see if the account exists and if it doesn't, we redirect
+ * users to `/signup`.
+ *
+ * In the React version, we're temporarily always passing the `email` param over
+ * from the Backbone index page until the index page is converted over, in which case
+ * we can pass the param with router state. Since we already perform this account exists
+ * (account status) check on the Backbone index page, which is rate limited since it doesn't
+ * require a session token, we also temporarily pass email status params to 1) signal not to
+ * perform the check again but also because 2) these params are needed to conditionally
+ * display UI in signin. If no status params are passed and `email` is, or we read the
+ * email from local storage, we perform the check and redirect existing user emails to
+ * `/signup` to match content-server functionality.
+ */
+
+const SigninContainer = ({
```

Comment:
Are container tests going to be added in a follow up?

Reply:
Depends on how much I get done tomorrow, but the answer is probably 😅